### PR TITLE
Accessing store_type column from volume table

### PR DESCRIPTION
**Pull request by RakeshAMore**
Problem:
Need provision to access store_type column from volume table.

Solution:
Modified related classes to access store_type column. We will be modifying these classes again for updating to create a query as part of [ZCS-11428](https://jira.corp.synacor.com/browse/ZCS-11428) and [ZCS-11430](https://jira.corp.synacor.com/browse/ZCS-11430)

### CODE REVIEW

File: store/src/java/com/zimbra/cs/volume/Volume.java [hunk 1]
```diff
@@ -65,7 +65,30 @@ public final class Volume {
...
+
+        private final int storeType;
+        StoreType(final int storeType) {
+            this.storeType = storeType;
+        }
+
+        public int getStoreType() {
+            return storeType;
+        }
+        public static StoreType getStoreTypeBy(int value){
```
Comment (deepakzimb) [inline store/src/java/com/zimbra/cs/volume/Volume.java:83]: please give space before {

**Review by deepakzimb — APPROVED** [review]

File: store/src/java/com/zimbra/cs/db/DbVolume.java [hunk 2]
```diff
@@ -329,15 +336,15 @@ public static boolean isVolumeReferenced(DbConnection conn, short volumeId) thro
...
+                tableRefsVolume(conn, volumeId, groupId, "mail_item_dumpster") || tableRefsVolume(conn, volumeId, groupId, "mail_item");
     }
 
     private static boolean tableRefsVolume(DbConnection conn, short volumeId, int groupId, String table) throws ServiceException {
         PreparedStatement stmt = null;
         ResultSet rs = null;
         try {
             stmt = conn.prepareStatement("SELECT count(*) from "+DbMailbox.qualifyTableName(groupId, table) +
-                " where locator=?");
+                    " where locator=?");
```
Comment (shrutig0510) [inline store/src/java/com/zimbra/cs/db/DbVolume.java:347]: Indentation issue.

File: store/src/java/com/zimbra/cs/db/DbVolume.java [hunk 3]
```diff
@@ -329,15 +336,15 @@ public static boolean isVolumeReferenced(DbConnection conn, short volumeId) thro
 
     private static boolean isVolumeReferenced(DbConnection conn, short volumeId, int groupId) throws ServiceException {
         return tableRefsVolume(conn, volumeId, groupId, "revision") || tableRefsVolume(conn, volumeId, groupId, "revision_dumpster") ||
-            tableRefsVolume(conn, volumeId, groupId, "mail_item_dumpster") || tableRefsVolume(conn, volumeId, groupId, "mail_item");
+                tableRefsVolume(conn, volumeId, groupId, "mail_item_dumpster") || tableRefsVolume(conn, volumeId, groupId, "mail_item");
```
Comment (shrutig0510) [inline store/src/java/com/zimbra/cs/db/DbVolume.java:339]: Indentation issue

Comment (RakeshAMore) [reply store/src/java/com/zimbra/cs/db/DbVolume.java:339]: I think this is the right, split statement should always be right to the above keyword.

File: store/src/java/com/zimbra/cs/db/DbVolume.java (hunk 2 above)
Comment (RakeshAMore) [reply store/src/java/com/zimbra/cs/db/DbVolume.java]: this is the same. but now moved it in single statement.

**Review by zimsuchitgupta — APPROVED** [review]

**Review by abhishekdeshmukh2209 — APPROVED** [review]

**Review by preshitaagrawal — APPROVED** [review]

**Pull request closed without merging**